src/auth/auth.ts

```diff
@@ -1,188 +1,45 @@
-    if (credentialsMethods.indexOf(config.method) != -1) {
-      if (!credentials) {
-        return this._reject('You must include credentials to use this auth method.');
-      }
-    }
-
-    switch (config.method) {
-      case AuthMethods.Popup:
-        return this._authBackend.authWithOAuthPopup(config.provider, this._scrubConfig(config))
-          .then((userCredential: firebase.auth.UserCredential) => {
-            // Incorrect type information
-            this._credentialCache[userCredential.credential.provider] = userCredential.credential;
-            return authDataToAuthState(userCredential.user, (<any>userCredential).credential);
-          });
-      case AuthMethods.Redirect:
-        // Gets around typings issue since this method doesn't resolve with a user.
-        // The method really only does anything with an error, since it redirects.
-        return <Promise<FirebaseAuthState>>(<any>this._authBackend).authWithOAuthRedirect(config.provider, this._scrubConfig(config));
-      case AuthMethods.Anonymous:
-        return this._authBackend.authAnonymously(this._scrubConfig(config));
-      case AuthMethods.Password:
-        return this._authBackend.authWithPassword(<EmailPasswordCredentials>credentials);
-      case AuthMethods.OAuthToken:
-        return this._authBackend.authWithOAuthToken(<firebase.auth.AuthCredential>credentials,
-          this._scrubConfig(config));
-      case AuthMethods.CustomToken:
-        return this._authBackend.authWithCustomToken(<string>credentials);
-    }
-  }
-
-  public logout(): Promise<void> {
-    return this._authBackend.unauth();
-  }
-
-  public getAuth(): FirebaseAuthState {
-    console.warn(`WARNING: the getAuth() API has changed behavior since adding support for Firebase 3.
-    This will return null for the initial value when the page loads, even if the user is actually logged in.
-    Please observe the actual authState asynchronously by subscribing to the auth service: af.auth.subscribe().
-    The getAuth method will be removed in future releases`);
-    return this._authBackend.getAuth()
-  }
-
-  public createUser(credentials: EmailPasswordCredentials): firebase.Promise<FirebaseAuthState> {
-    return this._authBackend.createUser(credentials);
-  }
+export class AngularFireAuth {
 
   /**
-   * Merges the config object that is passed in with the configuration
-   * provided through DI. Giving precendence to the one that was passed.
+   * Firebase Auth instance
    */
-  private _mergeConfigs(config: AuthConfiguration): AuthConfiguration {
-    if (this._config == null)
-      return config;
-
-    return Object.assign({}, this._config, config);
-  }
-
-  private _reject(msg: string): firebase.Promise<FirebaseAuthState> {
-    return (<Promise<FirebaseAuthState>>new Promise((res, rej) => {
-      return rej(msg);
-    }));
-  }
+  auth: firebase.auth.Auth;
+  
+  /**
+   * Observable of authentication state
+   */
+  authState: Observable<firebase.User>;
```

Comment:
@cartant I'm not so sure, it is an observable of user. But I guess that confusion could be that it can be null, and the null value is useful itself. That makes `authState` more suitable. @jaysol mentioned using just `.state()`. Thoughts?